File: src/net/dinkla/raytracer/brdf/FresnelReflector.java
package net.dinkla.raytracer.brdf;

import net.dinkla.raytracer.colors.Color;
import net.dinkla.raytracer.hits.Shade;
import net.dinkla.raytracer.math.Vector3DF;

/**
 * Created by IntelliJ IDEA.
 * User: Jörn Dinkla
 * Date: 15.05.2010
 * Time: 11:17:10
 * To change this template use File | Settings | File Templates.
 */
public class FresnelReflector<C extends Color> extends BRDF<C> {

    @Override
    public C f(final Shade sr, final Vector3DF wo, final Vector3DF wi){
        return null;
    }

    @Override
    public C rho(Shade sr, Vector3DF wo) {
        return null;
    }

    @Override
    public Sample sampleF(Shade sr, Vector3DF wo) {
        return null;
    }
}

File: src/main/kotlin/net/dinkla/raytracer/brdf/FresnelReflector.kt
package net.dinkla.raytracer.brdf

import net.dinkla.raytracer.colors.Color
import net.dinkla.raytracer.hits.Shade
import net.dinkla.raytracer.math.Vector3D

class FresnelReflector : BRDF() {
    override fun f(sr: Shade, wo: Vector3D, wi: Vector3D): Color = Color.BLACK

    override fun rho(sr: Shade, wo: Vector3D): Color = Color.BLACK

    // TODO def
    override fun sampleF(sr: Shade, wo: Vector3D): BRDF.Sample = Sample(
            wi = Vector3D.ZERO, color = Color.BLACK, pdf = 1.0)
}
